src/gallery.tsx

```diff
@@ -370,6 +451,16 @@ export const Gallery: FC<GalleryProps> = ({
     }
   }, [open, galleryUID])
 
+  useEffect(() => {
+    openGalleryBasedOnUrlHash()
+
+    // needed for case when gallery was firstly opened, then was closed and user clicked on browser's forward button
+    window.addEventListener('popstate', openGalleryBasedOnUrlHash)
+    return () => {
+      window.removeEventListener('popstate', openGalleryBasedOnUrlHash)
+    }
```

Comment:
To keep separation of concerns I think should be split out into its own `useEffect`.